src/machine/machine_stm32_i2c.go

```diff
@@ -274,13 +252,7 @@ func (i2c I2C) controllerRequestWrite(addr addressable, option transferOption) e
 	}
 
 	// send peripheral address
-	switch addr.bitSize() {
-	case 7: // 7-bit peripheral address
-		i2c.Bus.DR.Set(addr.toWrite())
-
-	case 10: // 10-bit peripheral address
-		// TODO
-	}
+	i2c.Bus.DR.Set(directionWrite.shift7Bit(addr))
```

Comment:
Personally I'd simplify this further by avoiding the indirection for what is really just a oneliner:

```suggestion
	i2c.Bus.DR.Set(uint32(addr) << 1)
```

The I2C protocol is fixed, this format is not going to change. Therefore I don't see the potential flexibility of moving this in a helper function.

Reply:
Replaced with inline shift, removed type and constants for `i2cDirection`